src/s3vfs.ts

```diff
@@ -126,8 +132,13 @@ export class S3VFS extends Base {
     return this.handleAsync(async () => {
       try {
         const objects = await this.fetchObjects(fileId)
-        const size = (objects || []).reduce((size, { Size: blockSize }) => size + (blockSize || 0), 0)
-        pSize64.set(size)
+        // Capture the last non-slash characters
+        const filename = objects?.[0]?.Key?.match(/([^/]*)$/)?.[0];
+        // console.log(filename);
+        // Use BigInt to avoid precision problems with Number
+        const fileCount = filename ? base36.decodeBigInt(filename) : 0;
+        const size = fileCount ? (fileCount * this.blockSizeN) + this.blockSizeN : 0;
+        pSize64.set(Number(size))
```

Comment:
For the avoidance of doubt, could you explain what is happening here?

afaict, when S3 responds with the list of files, and per #1, the lowest (and hence most recent in sequence, since we are using a descending sequence from `zzzzzzzzzz` to `0000000000`) filename will be assigned to `filename`. `size` will then be calculated from that filename. 

This should then result in `xFileSize()` whose value _decreases_ as the actual file size _increases_. Or am I mistaken here?

Reply:
You're correct. I need to reverse the offset here to get the difference between the largest and smallest filename. I'll add a commit